src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_FACTOR = 2;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
+
+    private static class Node<K, V> {
+        private K key;
+        private V value;
+        private int hash;
+        private Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next, int hash) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+            this.hash = hash;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            resize();
+        }
+        int hash = getHashCode(key);
+        int index = hash % table.length;
+        Node<K, V> newNode = new Node<>(key, value, null, hash);
+        Node<K, V> node = table[index];
+        if (node == null) {
+            table[index] = newNode;
+        }
+        while (node != null) {
+            if (node.hash == hash && Objects.equals(key, node.key)) {
+                node.value = value;
+                return;
+            }
+            if (node.next == null) {
+                node.next = newNode;
+                break;
+            }
+            node = node.next;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getHashCode(key) % table.length;
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (node.key == key || node.key != null && node.key.equals(key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        size = 0;
+        Node<K, V>[] oldTable = table;
+        table = new Node[table.length * RESIZE_FACTOR];
+        threshold *= RESIZE_FACTOR;
+        for (Node<K,V> node : oldTable) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
+
+    private int getHashCode(K key) {
```

Comment:
```suggestion
    private int getIndex(K key) {
```
you are getting index from this method